src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/DiscardParameterHighlighter.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editor.Implementation.Highlighting;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.Highlighting.KeywordHighlighters
+{
+    [ExportHighlighter(LanguageNames.CSharp)]
+    internal class DiscardParameterHighlighter : AbstractKeywordHighlighter<ArgumentSyntax>
+    {
+        [ImportingConstructor]
+        public DiscardParameterHighlighter()
+        {
+        }
+
+        protected override IEnumerable<TextSpan> GetHighlights(ArgumentSyntax node, CancellationToken cancellationToken)
+        {
+            if (node.Expression is IdentifierNameSyntax nameSyntax
+                && nameSyntax.Identifier.Text == "_")
```

Comment:
💡 Can use pattern matching here:

```
if (node.Expression is IdentifierNameSyntax { Identifier: { Text: "_" } } nameSyntax)
```

Reply:
I still need to get used to pattern matching tbh. Should all new code aim to make use of pattern matching or is it up to preference?

Reply:
It's up to preference.  in some places it works nicely (like here), in others, it can feel more forced/unpleasant.